src/components/bottom-bar-item/styles.ts

```diff
@@ -0,0 +1,82 @@
+import { styled } from '@linaria/react'
+import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'
+
+type CommonBottomBarItemProps = {
+  /**
+   * NOTE: data attribute to indicate the state of the nav item
+   * by default the styles is already applied with pseudo-class selector
+   * but to make it more flexible, consumer could use this attribute to
+   * change the styles
+   */
+  'data-state'?: 'active'
+}
+
+export const ElBottomBarItemIcon = styled.div`
+  * {
+    width: var(--icon-default, 24px);
+    height: var(--icon-default, 24px);
+    color: var(--icon-secondary, #607890);
+  }
+`
+
+export const ElBottomBarItemLabel = styled.span`
+  color: var(--text-secondary, #607890);
+  text-align: center;
+
+  font-family: var(--font-family, Inter);
+  font-size: var(--font-size-2xs, 10px);
+  font-style: normal;
+  font-weight: 500;
+  line-height: var(--line-height-2xs, 12px); /* 120% */
+  letter-spacing: var(--letter-spacing-2xs, 0px);
+`
+
+const baseStyles = `
+  background-color: var(--white, #ffffff);
+  outline: none;
+  border: none;
+  display: flex;
+  cursor: pointer;
+  padding: var(--spacing-half, 2px) var(--spacing-none, 0px);
+  flex-direction: column;
+  align-items: center;
+  gap: var(--spacing-half, 2px);
+  
+
+`
+
+interface ElAnchorBottomBarItemProps extends AnchorHTMLAttributes<HTMLAnchorElement>, CommonBottomBarItemProps {}
+export const ElAnchorBottomBarItemContainer = styled.a<ElAnchorBottomBarItemProps>`
+  ${baseStyles}
```

Comment:
**question:** why are the "active" styles not part of the `baseStyles`?

Reply:
>  why are the "active" styles not part of the baseStyles?

this is the interesting part. I attempted to incorporate the variable selector into the `baseStyles` using string interpolation from Linaria, but the expected node doesn't seem to have the correct CSS selector

as an example, assuming I've already declared a styled component named `ElBottomBarItemIcon` and need to modify its style based on the parent node's state, the first example works because it's plain text rendered as regular CSS. The generated Linaria class name, `.el-bottom-bar-item-icon`, directly affects the styles.

however, the second example doesn't work. Even after trying to escape the `$` symbol, the `styled` API from Linaria doesn't seem to process the style for that node. It appears that the node isn't being referenced correctly🫨

this works because it's directly translated to CSS
```ts
const baseStyles = `
  background-color: var(--fill-white, white);

  .el-bottom-bar-item-icon {
    color: var(--icon-secondary, #607890);
  }
`
```
here, the intent is to dynamically target the `ElBottomBarItemIcon` component, but the Linaria `styled` API doesn't seem to handle this correctly
```ts
const baseStyles = `
  background-color: var(--fill-white, white);

 ${ElBottomBarItemIcon} { // or \${ElBottomBarItemIcon}
    color: var(--icon-secondary, #607890);
  }
`
```

Reply:
**issue:** I think hardcoding the styled element's class name will present a lot of problems long-term:

1. It's not easily maintainable. If we change the name of the styled component, we may easily miss the need to also update the CSS where we have hardcoded its class name;
2. It's not obvious to a reader that the hardcoded class name is related to the styled component (certainly not to someone new to the repo/Linaria);
3. If we change the way Linaria formats the class names for styled components, these styles will break without us knowing.

Instead of needing to change the colour of the subcomponents, why don't we just change the colour of the root element and make sure the subcomponents inherit that colour?

i.e. 
```javascript
const baseStyles = `
  color: ...;

  [aria-current] {
    color: ...;
  }
`

const ElBottomBarItemIcon = styled.div`
  color: inherit;
`

const ElBottomBarItemLabel = styled.div`
  color: inherit;
`
```

Reply:
yes, hardcoding is fragile in the long term. Your suggestion seems understandable; the subcomponents could inherit. I totally agree